app/database/operator/server_data_operator/handlers/post.ts

```diff
@@ -104,6 +108,92 @@ export const exportedForTest = {
 };
 
 const PostHandler = <TBase extends Constructor<ServerDataOperatorBase>>(superclass: TBase) => class extends superclass {
+    /**
+     * handleScheduledPosts: Handler responsible for the Create/Update operations occurring the SchedulePost table from the 'Server' schema
+     * @param {HandleScheduledPostsArgs} ScheduledPostsArgs
+     * @param {boolean} draftsArgs.prepareRecordsOnly
+     * @returns {Promise<Model[]>}
+     */
+    handleScheduledPosts = async ({actionType, scheduledPosts, prepareRecordsOnly}: HandleScheduledPostsArgs): Promise<ScheduledPostModel[]> => {
+        const database: Database = this.database;
+
+        if (!scheduledPosts?.length && actionType === ActionType.SCHEDULED_POSTS.RECEIVED_ALL_SCHEDULED_POSTS) {
+            await database.write(async () => {
+                const scheduledPostCollection = database.get<ScheduledPostModel>(SCHEDULED_POST);
+                await scheduledPostCollection.query().destroyAllPermanently();
+            });
+            return [];
+        }
+
+        if (!scheduledPosts?.length) {
+            logWarning(
+                'An empty or undefined "scheduledPosts" array has been passed to the handleScheduledPosts method',
+            );
+            return [];
+        }
+
+        if (actionType === ActionType.SCHEDULED_POSTS.DELETE_SCHEDULED_POST) {
+            const deletedScheduledPostIds = scheduledPosts.map((post) => post.id);
+            await this._deleteScheduledPostByIds(deletedScheduledPostIds);
+            return [];
+        }
+
+        const createOrUpdateRawValues = getUniqueRawsBy({raws: scheduledPosts, key: 'id'}) as ScheduledPost[];
+
+        if (actionType === ActionType.SCHEDULED_POSTS.CREATE_OR_UPDATED_SCHEDULED_POST) {
+            return this._createOrUpdateScheduledPost(createOrUpdateRawValues, prepareRecordsOnly);
+        }
+
+        const idsFromServer = new Set(scheduledPosts.map((post) => post.id));
+        const existingScheduledPosts = await database.get<ScheduledPostModel>(SCHEDULED_POST).query().fetch();
+
+        const deletedScheduledPostIds = existingScheduledPosts.filter((post) => !idsFromServer.has(post.id)).reduce((result, post) => {
+            result.add(post.id);
+            return result;
+        }, new Set<string>());
+
+        if (deletedScheduledPostIds.size) {
+            this._deleteScheduledPostByIds(Array.from(deletedScheduledPostIds));
```

Comment:
We need to return all modified models, in case we are passing `prepareRecordsOnly`. So in this case, we should instead of calling in `_deleteScheduledPostByIds` `destroyPermanently` we should just prepare the deletion, and then batch all the models together if we shouldn't prepare the records only.

Reply:
Understood. Will update the code.